### PR TITLE
Outstanding TODOs from PR #51

### PR DESCRIPTION
**Pull request by ckotzbauer**
- Updated standard-version to 4.0.0 and set tagPrefix to empty string
- Extracted cli stuff to own file from index.js

In Addition to #51 

@EisenbergEffect @niieani

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
@jdanyow @niieani Can one of you guys take a look at this?

File: src/index.js [hunk 1]
```diff
@@ -1,22 +1,9 @@
-#!/usr/bin/env node
-"use strict";
```
Comment (niieani) [inline src/index.js]: We probably wanna keep "use strict" here

**Comment by niieani** [comment]
Does this fix the doc building issue on Windows?

**Comment by StrahilKazlachev** [comment]
@niieani No, I still get
```
> aurelia-tools doc-build --continue-when-failed

events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: spawn ./node_modules/.bin/tsc ENOENT
    at exports._errnoException (util.js:1022:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:193:32)
    at onErrorNT (internal/child_process.js:359:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at Module.runMain (module.js:606:11)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
```
This [line](https://github.com/aurelia/tools/blob/cli/src/cli.js#L111) is causing the issue.

EDIT - What I did:
1. I checked out `loader-webpack`
2. `npm install`
3. `npm run build`

**Comment by ckotzbauer** [comment]
I can check this today evening

**Comment by ckotzbauer** [comment]
@StrahilKazlachev @niieani 
This can be fixed if this line
```
const tsc = spawn( './node_modules/.bin/tsc', [ '--project', argv.project, '--outFile', path.join(argv.outDir, packageName + '.js') ] );
```
is replaced with this line:
```
const tsc = spawn( "node", [ './node_modules/typescript/bin/tsc', '--project', argv.project, '--outFile', path.join(argv.outDir, packageName + '.js') ] );
```

The `typedoc` subcommand also has problems, because it only prints the usage docs...

**Comment by ckotzbauer** [comment]
Ok, typedoc seems to have no problems. I used the command in a wrong way. I can submit a PR for the first problem with doc-build, if you want.

**Comment by niieani** [comment]
@code-chris that would be fantastic. Please target the #cli branch.

**Comment by ckotzbauer** [comment]
@niieani done #56

**Comment by StrahilKazlachev** [comment]
@code-chris Nice!
@niieani I had to explicitly install typedoc @ 0.5.1. The implicitly installed 0.5.5 decided that it doesn't like some of the options, it spit:
```
Usage:
 typedoc --mode modules --out path/to/documentation path/to/sourcefiles
TypeDoc options:
 --disableOutputCheck                Should TypeDoc disable the testing and cleaning of the output directory?
 --entryPoint                        Specifies the fully qualified name of the root symbol. Defaults to global namespace.
 --exclude                           Define a pattern for excluded files when specifying paths.
 --excludeExternals                  Prevent externally resolved TypeScript files from being documented.
 --excludeNotExported                Prevent symbols that are not exported from being documented.
 ... and so on
```
```
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "bla\\bla\\bla\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "build:doc"
npm ERR! node v6.9.4
npm ERR! npm  v4.1.2
npm ERR! code ELIFECYCLE
npm ERR! aurelia-loader-webpack@2.0.0 build:doc: `aurelia-tools doc-shape-defs && aurelia-tools typedoc`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the aurelia-loader-webpack@2.0.0 build:doc script 'aurelia-tools doc-shape-defs && aurelia-tools typedoc'.
```
With 0.5.1 the TS just complains `Cannot find name 'Promise'.`, but it finishes.

**Comment by niieani** [comment]
Hmm, thanks for reporting @StrahilKazlachev. We should probably do something about that. Again, open to PRs with fixes.

**Pull request merged**